astropy/__init__.py

```diff
@@ -205,48 +190,24 @@ class astronomical_constants(base_constants_version):
 # if we are *not* in setup mode, import the logger and possibly populate the
 # configuration file with the defaults
 def _initialize_astropy():
-    from . import config
-
-    def _rollback_import(message):
-        log.error(message)
-        # Now disable exception logging to avoid an annoying error in the
-        # exception logger before we raise the import error:
-        _teardown_log()
-
-        # Roll back any astropy sub-modules that have been imported thus
-        # far
 
-        for key in list(sys.modules):
-            if key.startswith('astropy.'):
-                del sys.modules[key]
-        raise ImportError('astropy')
+    from . import config
 
     try:
         from .utils import _compiler
     except ImportError:
         if _is_astropy_source():
-            log.warning('You appear to be trying to import astropy from '
-                        'within a source checkout without building the '
-                        'extension modules first.  Attempting to (re)build '
-                        'extension modules:')
-
-            try:
-                _rebuild_extensions()
-            except BaseException as exc:
-                _rollback_import(
-                    'An error occurred while attempting to rebuild the '
-                    'extension modules.  Please try manually running '
-                    '`./setup.py build_ext --inplace` to see what the issue was. Extension '
-                    'modules must be successfully compiled and importable '
-                    'in order to import astropy.')
-                # Reraise the Exception only in case it wasn't an Exception,
-                # for example if a "SystemExit" or "KeyboardInterrupt" was
-                # invoked.
-                if not isinstance(exc, Exception):
-                    raise
-
+            raise ImportError('You appear to be trying to import astropy from '
+                              'within a source checkout or from an editable '
+                              'installation without building the extension '
+                              'modules first. Either run:\n\n'
+                              '  pip install -e .\n\nor\n\n'
+                              '  python setup.py build_ext --inplace\n\n'
+                              'to make sure the extension modules are built '
+                              '(note that if you use the latter you may need '
```

Comment:
```suggestion
                              '(note that if you use the latter you need '
```
It's more than a supposition right?

Reply:
I went with @eteq's suggestion which was even more drastic